massa-final-state/src/final_state.rs

```diff
@@ -151,6 +151,10 @@ impl FinalState {
         // 6. final state hash
         let final_state_hash = Hash::compute_from(&hash_concat);
         info!("final_state hash at slot {}: {}", slot, final_state_hash);
+        // 7. save final state hash in the latest cycle
+        if let Some(cycle) = self.pos_state.cycle_history.back_mut() {
```

Comment:
a more robust and less error-prone way would be to create a feed_cycle_state_hash function in the pos state, and have it internally use get_cycle_index to make sure it grabs the right state